src/DataTable/index.jsx

```diff
@@ -111,10 +111,12 @@ function DataTable({
   // any state changes to current row selections as we don't want to re-fetch data whenever row(s) are selected.
   const tableStateWithoutSelections = { ...instance.state };
   delete tableStateWithoutSelections.selectedRowIds;
+
   useEffect(() => {
     if (fetchData) {
       fetchData(tableStateWithoutSelections);
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
The point is that in the comment above the `tableStateWithoutSelections`:

> Call ``fetchData`` whenever the state of the table changes (e.g., page change, sort or filter applied) but ignore

This means we want this `useEffect` to work every time the `instance.state` changes. If we just pass `tableStateWithoutSelections` without `JSON.stringify`, any variables changes inside won't be detected due to not deep comparison of `Objects`. The same thing will be for `useMemo` because we would need `JSON.stringify` for `instance.state`. Otherwise changes to state won't trigger changes in the `tableStateWithoutSelections` variable.